benches/benches/block_target_gas_set/crypto.rs

```diff
@@ -0,0 +1,171 @@
+            .collect(),
+    );
+
+    let wrong_message = Message::new(b"bar");
+
+    run(
+        "crypto/eck1 opcode invalid",
+        group,
+        [
+            op::gtf_args(0x20, 0x00, GTFArgs::ScriptData),
+            op::addi(
+                0x21,
+                0x20,
+                eck1_signature.as_ref().len().try_into().unwrap(),
+            ),
+            op::movi(0x10, PublicKey::LEN.try_into().unwrap()),
+            op::aloc(0x10),
+            op::eck1(RegId::HP, 0x20, 0x21),
+            op::jmpb(RegId::ZERO, 0),
+        ]
+        .to_vec(),
+        eck1_signature
+            .iter()
+            .chain(wrong_message.iter())
+            .copied()
+            .collect(),
+    );
+
+    let message = fuel_core_types::fuel_crypto::Message::new(b"foo");
+    let ecr1_secret = p256::ecdsa::SigningKey::random(&mut rand::thread_rng());
+    let ecr1_signature = secp256r1::sign_prehashed(&ecr1_secret, &message)
+        .expect("Failed to sign with secp256r1");
+
+    run(
+        "crypto/ecr1 opcode",
+        group,
+        [
+            op::gtf_args(0x20, 0x00, GTFArgs::ScriptData),
+            op::addi(
+                0x21,
+                0x20,
+                ecr1_signature.as_ref().len().try_into().unwrap(),
+            ),
+            op::movi(0x10, PublicKey::LEN.try_into().unwrap()),
+            op::aloc(0x10),
+            op::move_(0x11, RegId::HP),
+            op::ecr1(0x11, 0x20, 0x21),
+            op::jmpb(RegId::ZERO, 0),
+        ]
+        .to_vec(),
+        ecr1_signature
+            .as_ref()
+            .iter()
+            .chain(message.as_ref())
+            .copied()
+            .collect(),
+    );
+
+    let message = fuel_core_types::fuel_crypto::Message::new(b"foo");
+    let ed19_keypair =
+        ed25519_dalek::Keypair::generate(&mut ed25519_dalek_old_rand::rngs::OsRng {});
+    let ed19_signature = ed19_keypair.sign(&*message);
+
+    run(
+        "crypto/ed19 opcode",
+        group,
+        [
+            op::gtf_args(0x20, 0x00, GTFArgs::ScriptData),
+            op::addi(
+                0x21,
+                0x20,
+                ed19_keypair.public.as_ref().len().try_into().unwrap(),
+            ),
+            op::addi(
+                0x22,
+                0x21,
+                ed19_signature.as_ref().len().try_into().unwrap(),
+            ),
+            op::addi(0x22, 0x21, message.as_ref().len().try_into().unwrap()),
+            op::movi(0x10, ed25519_dalek::PUBLIC_KEY_LENGTH.try_into().unwrap()),
+            op::aloc(0x10),
+            op::move_(0x11, RegId::HP),
+            op::ed19(0x20, 0x21, 0x22),
+            op::jmpb(RegId::ZERO, 0),
+        ]
+        .to_vec(),
+        ed19_keypair
+            .public
+            .as_ref()
+            .iter()
+            .chain(ed19_signature.as_ref())
+            .chain(message.as_ref())
+            .copied()
+            .collect(),
+    );
+
+    for i in generate_linear_costs() {
```

Comment:
I understand you didn't create `generate_linear_costs()` but I'm curious to know why it's being used here. It looks like it generates `n`s that increase exponentially and then decrease exponentially, and for each `n`, we run `sha256` on `n` bytes. Is there a reason that we use this specific generator function over something like a monotonic or linear generator?

Reply:
I don't really understand the name--I've been trying to think of a new name. But it's used in a number of or benchmarks for dependent costs. The actual values used aren't particularly important as long as they show a good range between 0 -> a lot.

Reply:
Renaming `arb_dependent_cost_values` to communicate that they aren't particularly important values and what they are for.